.circleci/config.yml

```diff
@@ -1,111 +1,111 @@
+            #            - v2-npm-deps-
             - run:
-                name: upload coverage report
-                command: |
-                  npx codecov -f packages/core/coverage/lcov.info -F core
-                  npx codecov -f packages/http-server/coverage/lcov.info -F http-server
-                  npx codecov -f packages/http-server-express/coverage/lcov.info -F http-server-express
-                  npx codecov -f packages/reflector/coverage/lcov.info -F reflector
-            - save_cache:
-                key: v1-source-{{ .Environment.CIRCLE_SHA1 }}
-                paths:
-                  - "~"
-  publish:
-    docker:
-      - image: cimg/node:16.18.1
-    steps:
-      - restore_cache:
-          key: v1-source-{{ .Environment.CIRCLE_SHA1 }}
-      - add_ssh_keys
-      - run:
-          name: Add github to known_hosts
-          command: |
-            mkdir -p ~/.ssh
-            ssh-keyscan github.com >> ~/.ssh/known_hosts
-      - run:
-          name: publish
-          command: |
-            git config --global user.email "circleci@hp.com"
-            git config --global user.name "CircleCI"
-            git checkout -f
-            npm run semantic-release
+                  name: install NPM dependencies
+                  command: |
+                      echo -e "//registry.npmjs.org/:_authToken=$NPM_TOKEN\nscope=@davinci" > .npmrc
+                      npm install
+            #      - save_cache:
+            #          key: v2-npm-deps-{{ .Branch }}-{{ checksum "package.json" }}-{{ checksum "package-lock.json" }}
+            #          paths:
+            #            - node_modules
+            - run:
+                  name: build
+                  command: |
+                      npm run build
+            - run:
+                  name: test
+                  command: |
+                      npm run cover
+            - when:
+                  condition:
+                      and:
+                          - equal: ['5', << parameters.mongo-version >>]
+                          - equal: ['cimg/node:18.16.0', << parameters.docker-image >>]
+                  steps:
+                      - run:
+                            name: upload coverage report
+                            command: |
+                                npx codecov -f packages/core/coverage/lcov.info -F core
+                                npx codecov -f packages/http-server/coverage/lcov.info -F http-server
+                                npx codecov -f packages/http-server-express/coverage/lcov.info -F http-server-express
+                                npx codecov -f packages/reflector/coverage/lcov.info -F reflector
+                      - save_cache:
+                            key: v1-source-{{ .Environment.CIRCLE_SHA1 }}
+                            paths:
+                                - '~'
+    publish:
+        docker:
+            - image: cimg/node:16.18.1
```

Comment:
We should update EOL versions to new LTS' https://nodejs.org/en/about/previous-releases

Reply:
moved to 18 and 20